src/test/java/httpserver/HttpParserTest.java

```diff
@@ -0,0 +1,24 @@
+package httpserver;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class HttpParserTest {
+    @Test
+    void itParsesTheRequestLineOfAnHttpRequest() {
+        HttpParser httpParser = new HttpParser();
+        String requestLine = "GET /hello_world HTTP/1.1" + System.lineSeparator();
+        ArrayList<String> headers = new ArrayList<String>(Arrays.asList("Host: localhost:8000", "Connection: keep-alive"));
+        String allHeaders = String.join(System.lineSeparator(), headers) + System.lineSeparator();
+        String blankLine = System.lineSeparator();
+        String body = "Hello world";
+        String request = requestLine + allHeaders + blankLine + body;
+        String path = "/hello_world";
+
+        assertEquals(path, httpParser.parse(request));
```

Comment:
Maybe it is just me, but this test is really hard for me to follow. I think it might be because I have a hard time following the setup. It isn't clear what is important and what isn't.

My approach to this test would be to do something like the following

```java
    @Test
    void itParsesTheRequestLineOfAnHttpRequest() {
        HttpParser httpParser = new HttpParser();
        
        String request = (
                "GET /hello_world HTTP/1.1\r\n" +
                "Host: localhost:8000\r\n" +
                "Connection: keep-alive\r\n" +
                "\r\n" +
                "Hello world"
        );
        
        String path = "/hello_world";

        assertEquals(path, httpParser.parse(request));
    }
```

IMO this makes it really clear what the input to the `HttpParser` is.

Reply:
Thanks.  This is much clearer.